core/federated/RTI/rti_remote.c

```diff
@@ -250,30 +250,25 @@ void notify_provisional_tag_advance_grant(scheduling_node_t* e, tag_t tag) {
         // Send PTAG to all upstream federates, if they have not had
         // a later or equal PTAG or TAG sent previously and if their transitive
         // NET is greater than or equal to the tag.
+        // This is needed to stimulate absent messages from upstream and break deadlocks.
```

Comment:
It is a little strange that sending a PTAG to federate A should trigger PTAGs to all federates UPSTREAM of A? The federates upstream should not depend on A receiving a PTAG before advancing their time? Maybe we could explain with a few more words these deadlock scenarios that apparently can occur?

Reply:
I believe this is correct. An upstream federate may not have any events on its event queue to process at the tag of this PTAG.  If it stays that way, it will never send absent messages, thereby potentially causing a deadlock.  Sending a PTAG to the upstream federate causes it to put a dummy event on its event queue, which will stimulate absent messages at that tag even if there are no other events to process at that tag.